app/scripts/models/auth_brokers/oauth.js

```diff
@@ -102,6 +118,36 @@ define(function (require, exports, module) {
         .then(_formatOAuthResult);
     },
 
+    /**
+     * Derive scoped keys and encrypt them with the relier's public JWK
+     *
+     * @param {Object} account
+     * @param {String} assertion
+     * @returns {Promise} Returns a promise that resolves into an encrypted bundle
+     * @private
+     */
+    _provisionScopedKeys (account, assertion) {
+      const relier = this.relier;
+      const keyFetchToken = account.get('keyFetchToken');
+      const unwrapBKey = account.get('unwrapBKey');
+
+      // if check if requested scopes provide scoped keys
+      return this._oAuthClient.getClientKeyData({
+        assertion: assertion,
+        client_id: relier.get('clientId'), //eslint-disable-line camelcase
+        scope: decodeURIComponent(relier.get('scope'))
+      }).then((clientKeyData) => {
```

Comment:
Can the check for `unwrapBKey` and `keyFetchToken` be done before calling `getClientKeyData`? If the preconditions aren't met, no need to make an XHR request.

Reply:
👍